### PR TITLE
Kalnin Alexandr Lab1 Var 3

### PR DESCRIPTION
**Pull request by KalininA1**

### CODE REVIEW

File: clang/labs/lab1/kalinin_alexandr/deprWarnPluginKalinin.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  }
+};
+
+class PluginDeprFunc : public PluginASTAction {
+  std::unique_ptr<ASTConsumer>
+  CreateASTConsumer(CompilerInstance &Instance,
+                    llvm::StringRef InFile) override {
+    return std::make_unique<CustomConsumer>();
+  }
+  bool ParseArgs(const CompilerInstance &Compiler,
```
Comment (aobolensk) [inline clang/labs/lab1/kalinin_alexandr/deprWarnPluginKalinin.cpp:51]: Please add an option for this pass to support both case sensitive and case insensitive match

**Review by aobolensk — CHANGES_REQUESTED** [review]

File: clang/labs/lab1/kalinin_alexandr/deprWarnPluginKalinin.cpp (hunk 1 above)
Comment (KalininA1) [reply clang/labs/lab1/kalinin_alexandr/deprWarnPluginKalinin.cpp:51]: @aobolensk  
Thanks, I've made the changes

Comment (aobolensk) [reply clang/labs/lab1/kalinin_alexandr/deprWarnPluginKalinin.cpp:51]: I think tests are still needed

Comment (SSuren4ik) [reply clang/labs/lab1/kalinin_alexandr/deprWarnPluginKalinin.cpp:63]: Sorry, but I have tests

Comment (KalininA1) [reply clang/labs/lab1/kalinin_alexandr/deprWarnPluginKalinin.cpp:51]: @aobolensk 
Sorry, but I have tests. What did you mean?

Comment (aobolensk) [reply clang/labs/lab1/kalinin_alexandr/deprWarnPluginKalinin.cpp:51]: For both case sensitive and case insensitive

Comment (KalininA1) [reply clang/labs/lab1/kalinin_alexandr/deprWarnPluginKalinin.cpp:51]: @aobolensk tests added

**Review by aobolensk — APPROVED** [review]

**Pull request merged**